jira/src/getData.js

```diff
@@ -0,0 +1,159 @@
+      var fieldId = field.getId();
+      var fieldType = field.getType();
+      switch (fieldId) {
+        case 'id':
+          return row.push(issue.id);
+        case 'url':
+          return row.push(
+            'https://' + request.configParams.host + '/browse/' + issue.key
+          );
+        case 'issuekey':
+          return row.push(issue.key);
+        default:
+          var field = issue.fields[fieldId];
+          var result = '';
+          if (field) {
+            if (field.constructor === ''.constructor) {
+              // Process strings here
+              if (fieldType == 'YEAR_MONTH_DAY_HOUR') {
+                result = Utilities.formatDate(
+                  new Date(field),
+                  timeZone,
+                  format
+                );
+              } else {
+                result = field;
+              }
+            } else if (field.constructor === (0).constructor) {
+              // Process numbers here
+              result = field;
+            } else if (field.constructor === [].constructor) {
+              // Process arrays here
+              result = field.join();
+            } else if (field.constructor === {}.constructor) {
+              // Process object here
+              result =
+                field.displayName ||
+                field.value ||
+                field.name ||
+                JSON.stringify(field);
+            }
+          }
+          return row.push(result);
+      }
+    });
+    return {values: row};
+  });
+}
+
+/**
+ * Return true if valid
+ * @param {object} value
+ * @return {boolean} True if is not '', null or undefined
+ */
+function hasValue(value) {
+  return ['', null, undefined].indexOf(value) < 0;
+}
+
+/**
+ * Gets the data for the community connector
+ * @param {object} request The request.
+ * @return {object} The data.
+ */
+function getData(request) {
+  var requestedFieldIds = request.fields.map(function(field) {
+    return field.name;
+  });
+  var fieldsData = getJiraFields(request);
+  var requestedFields = getFields(fieldsData).forIds(requestedFieldIds);
+  var jql = [
+    request.configParams.dateForQuery != 'none'
+      ? request.configParams.dateForQuery +
+        ' >= ' +
+        request.dateRange.startDate +
+        ' AND ' +
+        request.configParams.dateForQuery +
+        ' <= ' +
+        request.dateRange.endDate
+      : '',
+    request.configParams.dateForQuery != 'none' ? 'AND ' : '',
+    hasValue(request.configParams.projects)
+      ? 'project in (' + request.configParams.projects + ')'
+      : '',
+    hasValue(request.configParams.projects) &&
+    hasValue(request.configParams.additionalQuery)
+      ? 'AND '
+      : '',
+    hasValue(request.configParams.additionalQuery)
+      ? request.configParams.additionalQuery
+      : ''
+  ];
+  var params = getParams();
+  var response = null;
+  var parsedResponse = null;
+  var startAt = 0;
+  var total = null;
+  var issues = [];
+  do {
+    var url = [
```

Comment:
There should be some error handling here in case the request fails for invalid credentials.